charms/argo-controller/charmcraft.yaml

```diff
@@ -12,3 +12,9 @@ bases:
 parts:
   charm:
     charm-python-packages: [setuptools, pip]  # Fixes install of some packages
+    # Following lines are needed due to https://github.com/canonical/charmcraft/issues/1722
+    build-snaps: [rustup]
+    build-packages: [pkg-config, libffi-dev, libssl-dev]
```

Comment:
Do we need to install these `build-packages` as well? For me I only did the following and it worked (but with Python 3.12):
```yaml
    build-snaps:
      - rustup
    override-build: |
      rustup default stable
      craftctl default
```
https://github.com/canonical/github-profiles-automator/pull/12/files#diff-8af267359854d30b091fde763088e458bf27c8fedc056895f3104ef05b61f4b5R24-R28

Reply:
Yes they are needed. Otherwise, building [fails](https://github.com/canonical/argo-operators/actions/runs/12199502801/job/34033534048#step:4:729) (example from [PR](https://github.com/canonical/argo-operators/pull/223) including both dependency upgrades + rust). FYI, the linked fix in the issue also includes the packages https://github.com/canonical/charmcraft/issues/1722#issuecomment-2259780356 which is what mysql-k8s has.